### PR TITLE
waitfor + checking for input c:

### PR DESCRIPTION
**Pull request by ItsRaijin**
rhythm i mean

### CODE REVIEW

File: src/minigames/rhythmMinigame.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { TextStyle, Text } from "pixi.js";
+import { Minigame } from "./minigame";
+
+export class RhythmMinigame extends Minigame {
```
Comment (zainafzal08) [inline src/minigames/rhythmMinigame.ts]: this is actually a pretty clean extension of the keyboardMinigame. Just extend it then override onKeyDown.

File: src/minigames/rhythmMinigame.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        await this.waitFor(500);
+        console.log("af");
+        this.waitingForPress = false;
+    }
+
+    async waitFor(time: number) {
+        return new Promise(resolve => setTimeout(resolve, time));
+    }
+
+
```
Comment (zainafzal08) [inline src/minigames/rhythmMinigame.ts]: why is there the pacific ocean of white space in this file?

File: src/minigames/rhythmMinigame.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        return new Promise(resolve => setTimeout(resolve, time));
+    }
+
+    
+
+    protected onKeyDown(key: string) {
+        console.log(key, "pressed");
+        if (this.waitingForPress == true) {
+            this.app.renderer.background.color = "#00FF00"
+            this.waitingForPress = false;
```
Comment (zainafzal08) [inline src/minigames/rhythmMinigame.ts:42]: Also just call this.delegate.onMinigameFinished(true) here and this.delegate.onMinigameFinished(false) in the else statement after a 1second delay. That way this minigame actually has a win and fail condition.

File: src/controller.ts [hunk 4]
```diff
@@ -1,9 +1,10 @@
 import { Application } from "pixi.js";
 import { Minigame, MinigameDelegate } from "./minigames/minigame";
 import { KeyboardMinigame } from "./minigames/keyboardMinigame";
+import { RhythmMinigame } from "./minigames/rhythmMinigame";
 import { Interlude, InterludeDelegate } from "./interlude/interlude";
 
-const ALL_MINIGAMES = [Minigame, KeyboardMinigame];
+const ALL_MINIGAMES = [RhythmMinigame];
```
Comment (zainafzal08) [inline src/controller.ts]: Just do the thing i suggested of actually ending the minigame and then add rhythmMinigame at the end of the list.

**Review by zainafzal08 — APPROVED** [review]

**Review by zainafzal08 — APPROVED** [review]

File: src/minigames/rhythmMinigame.ts [hunk 5]
```diff
@@ -1,22 +1,12 @@
 import { TextStyle, Text } from "pixi.js";
-import { Minigame } from "./minigame";
+import { KeyboardMinigame } from "./keyboardMinigame";
 
-export class RhythmMinigame extends Minigame {
+export class RhythmMinigame extends KeyboardMinigame {
 
-    private keyDownListener = (e: KeyboardEvent) => void this.onKeyDown(e.key);
+    //private keyDownListener = (e: KeyboardEvent) => void this.onKeyDown(e.key);
```
Comment (josephkimsungha) [inline src/minigames/rhythmMinigame.ts:6]: Pls remove

File: src/minigames/rhythmMinigame.ts [hunk 6]
```diff
@@ -46,16 +36,16 @@ export class RhythmMinigame extends Minigame {
...
     }
 
-    
-
-    protected onKeyDown(key: string) {
+    protected override onKeyDown(key: string) {
         console.log(key, "pressed");
         if (this.waitingForPress == true) {
             this.app.renderer.background.color = "#00FF00"
+            his.delegate.onMinigameFinished(true)
```
Comment (josephkimsungha) [inline src/minigames/rhythmMinigame.ts]: his -> this

File: src/minigames/rhythmMinigame.ts [hunk 7]
```diff
@@ -46,16 +36,16 @@ export class RhythmMinigame extends Minigame {
...
+    protected override onKeyDown(key: string) {
         console.log(key, "pressed");
         if (this.waitingForPress == true) {
             this.app.renderer.background.color = "#00FF00"
+            his.delegate.onMinigameFinished(true) 
             this.waitingForPress = false;
         }
         else {
             this.app.renderer.background.color = "#de3249"
+            his.delegate.onMinigameFinished(false)
```
Comment (josephkimsungha) [inline src/minigames/rhythmMinigame.ts]: his -> this

File: src/minigames/rhythmMinigame.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+    async countdown() {
+        await this.waitFor(1000);
+        this.text.text = "2";
+        await this.waitFor(1000);
+        this.text.text = "1";
+        await this.waitFor(1000);
+        this.text.text = "NOW";
+        this.waitingForPress = true;
+        console.log("bf");
```
Comment (josephkimsungha) [inline src/minigames/rhythmMinigame.ts]: Pls remove this and line 31

**Review by josephkimsungha — APPROVED** [review]

**Pull request merged**